File: CS/04_Graphs/List.cs
using System;
using System.Data.OleDb;
using System.Drawing;
using System.Windows.Forms;
using Spire.Pdf;
using Spire.Pdf.Graphics;
using Spire.Pdf.Lists;

namespace List
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            //Create a pdf document
            PdfDocument doc = new PdfDocument();

            //Margin
            PdfUnitConvertor unitCvtr = new PdfUnitConvertor();
            PdfMargins margin = new PdfMargins();
            margin.Top = unitCvtr.ConvertUnits(2.54f, PdfGraphicsUnit.Centimeter, PdfGraphicsUnit.Point);
            margin.Bottom = margin.Top;
            margin.Left = unitCvtr.ConvertUnits(3.17f, PdfGraphicsUnit.Centimeter, PdfGraphicsUnit.Point);
            margin.Right = margin.Left;

            //Create one page
            PdfPageBase page = doc.Pages.Add(PdfPageSize.A4, margin);

            float y = 10;

            //Title
            PdfBrush brush1 = PdfBrushes.Black;
            PdfTrueTypeFont font1 = new PdfTrueTypeFont(new Font("Arial", 16f, FontStyle.Bold), true);
            PdfStringFormat format1 = new PdfStringFormat(PdfTextAlignment.Center);
            page.Canvas.DrawString("Categories List", font1, brush1, page.Canvas.ClientSize.Width / 2, y, format1);
            y = y + font1.MeasureString("Categories List", format1).Height;
            y = y + 5;

            RectangleF rctg = new RectangleF(new PointF(0, 0), page.Canvas.ClientSize);
            PdfLinearGradientBrush brush
                = new PdfLinearGradientBrush(rctg, Color.Navy, Color.OrangeRed, PdfLinearGradientMode.Vertical);
            PdfFont font = new PdfFont(PdfFontFamily.Helvetica, 12f, PdfFontStyle.Bold);
            String formatted
                = "Beverages\nCondiments\nConfections\nDairy Products\nGrains/Cereals\nMeat/Poultry\nProduce\nSeafood";

            //Create a list
            PdfList list = new PdfList(formatted);
            list.Font = font;
            list.Indent = 8;
            list.TextIndent = 5;
            list.Brush = brush;

            //Draw the list on the page
            PdfLayoutResult result = list.Draw(page, 0, y);
            y = result.Bounds.Bottom;

            //Create another list
            PdfSortedList sortedList = new PdfSortedList(formatted);
            sortedList.Font = font;
            sortedList.Indent = 8;
            sortedList.TextIndent = 5;
            sortedList.Brush = brush;
            //Draw the list on the page
            PdfLayoutResult result2 = sortedList.Draw(page, 0, y);

            y = result2.Bounds.Bottom;
            PdfOrderedMarker marker1 = new PdfOrderedMarker(PdfNumberStyle.LowerRoman, new PdfFont(PdfFontFamily.Helvetica, 12f));
            PdfSortedList list2 = new PdfSortedList(formatted);
            list2.Font = font;
            list2.Marker = marker1;
            list2.Indent = 8;
            list2.TextIndent = 5;
            list2.Brush = brush;
            PdfLayoutResult result3 = list2.Draw(page, 0, y);
            y = result3.Bounds.Bottom;

            PdfOrderedMarker marker2 = new PdfOrderedMarker(PdfNumberStyle.LowerLatin, new PdfFont(PdfFontFamily.Helvetica, 12f));
            PdfSortedList list3 = new PdfSortedList(formatted);
            list3.Font = font;
            list3.Marker = marker2;
            list3.Indent = 8;
            list3.TextIndent = 5;
            list3.Brush = brush;
            list3.Draw(page, 0, y);

            //Save pdf file
            doc.SaveToFile("List.pdf");
            doc.Close();

            //Launch the Pdf file
            PDFDocumentViewer("List.pdf");
        }

        private void PDFDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

    }
}

File: VB/04_Graphs/List.vb
Imports Spire.Pdf
Imports Spire.Pdf.Graphics
Imports Spire.Pdf.Lists

Namespace List
    Partial Public Class Form1
        Inherits Form
        Public Sub New()
            InitializeComponent()
        End Sub
        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
            ' Create a new PdfDocument object
            Dim doc As New PdfDocument()

            ' Create a unit converter for converting measurement units
            Dim unitCvtr As New PdfUnitConvertor()

            ' Set margin
            Dim margin As New PdfMargins()
            margin.Top = unitCvtr.ConvertUnits(2.54F, PdfGraphicsUnit.Centimeter, PdfGraphicsUnit.Point)
            margin.Bottom = margin.Top
            margin.Left = unitCvtr.ConvertUnits(3.17F, PdfGraphicsUnit.Centimeter, PdfGraphicsUnit.Point)
            margin.Right = margin.Left

            ' Add a new page to the document with A4 size and specified margins
            Dim page As PdfPageBase = doc.Pages.Add(PdfPageSize.A4, margin)

            ' Initialize a variable for the vertical position
            Dim y As Single = 10

            ' Set brush and font for the title text
            Dim brush1 As PdfBrush = PdfBrushes.Black
            Dim font1 As New PdfTrueTypeFont(New Font("Arial", 16.0F, FontStyle.Bold), True)

            ' Set formatting options for the title text
            Dim format1 As New PdfStringFormat(PdfTextAlignment.Center)

            ' Draw the title "Categories List" in the center of the page
            page.Canvas.DrawString("Categories List", font1, brush1, page.Canvas.ClientSize.Width / 2, y, format1)

            ' Update the vertical position
            y = y + font1.MeasureString("Categories List", format1).Height
            y = y + 5

            ' Create a linear gradient brush for the list items
            Dim rctg As New RectangleF(New PointF(0, 0), page.Canvas.ClientSize)
            Dim brush As New PdfLinearGradientBrush(rctg, Color.Navy, Color.OrangeRed, PdfLinearGradientMode.Vertical)

            ' Set font for the list items
            Dim font As New PdfFont(PdfFontFamily.Helvetica, 12.0F, PdfFontStyle.Bold)

            ' Define the formatted string for the list items
            Dim formatted As String = "Beverages" & vbLf & "Condiments" & vbLf & "Confections" & vbLf & "Dairy Products" & vbLf & "Grains/Cereals" & vbLf & "Meat/Poultry" & vbLf & "Produce" & vbLf & "Seafood"

            ' Create a PdfList object with the formatted string
            Dim list As New PdfList(formatted)
            list.Font = font
            list.Indent = 8
            list.TextIndent = 5
            list.Brush = brush

            ' Draw the list on the page and get the layout result
            Dim result As PdfLayoutResult = list.Draw(page, 0, y)

            ' Update the vertical position
            y = result.Bounds.Bottom

            ' Create another PdfSortedList object with the same formatted string
            Dim sortedList As New PdfSortedList(formatted)
            sortedList.Font = font
            sortedList.Indent = 8
            sortedList.TextIndent = 5
            sortedList.Brush = brush

            ' Draw the sorted list on the page and get the layout result
            Dim result2 As PdfLayoutResult = sortedList.Draw(page, 0, y)

            ' Update the vertical position
            y = result2.Bounds.Bottom

            ' Create an ordered marker with lower Roman numerals
            Dim marker1 As New PdfOrderedMarker(PdfNumberStyle.LowerRoman, New PdfFont(PdfFontFamily.Helvetica, 12.0F))

            ' Create another PdfSortedList object with the same formatted string and the ordered marker
            Dim list2 As New PdfSortedList(formatted)
            list2.Font = font
            list2.Marker = marker1
            list2.Indent = 8
            list2.TextIndent = 5
            list2.Brush = brush

            ' Draw the list with ordered markers on the page and get the layout result
            Dim result3 As PdfLayoutResult = list2.Draw(page, 0, y)

            ' Update the vertical position
            y = result3.Bounds.Bottom

            ' Create an ordered marker with lower Latin letters
            Dim marker2 As New PdfOrderedMarker(PdfNumberStyle.LowerLatin, New PdfFont(PdfFontFamily.Helvetica, 12.0F))

            ' Create another PdfSortedList object with the same formatted string and the ordered marker
            Dim list3 As New PdfSortedList(formatted)
            list3.Font = font
            list3.Marker = marker2
            list3.Indent = 8
            list3.TextIndent = 5
            list3.Brush = brush

            ' Draw the list with ordered markers on the page
            list3.Draw(page, 0, y)

            ' Save the PDF file as "List.pdf"
            doc.SaveToFile("List.pdf")

            ' Close the document
            doc.Close()

            ' Launch the Pdf file
            PDFDocumentViewer("List.pdf")
        End Sub

        Private Sub PDFDocumentViewer(ByVal fileName As String)
            Try
                Process.Start(fileName)
            Catch
            End Try
        End Sub

    End Class
End Namespace
